cmd/prometheus/main.go

```diff
@@ -209,6 +209,11 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 				config.DefaultConfig.GlobalConfig.ScrapeProtocols = config.DefaultNativeHistogramScrapeProtocols
 				config.DefaultGlobalConfig.ScrapeProtocols = config.DefaultNativeHistogramScrapeProtocols
 				level.Info(logger).Log("msg", "Experimental native histogram support enabled. Changed default scrape_protocols to prefer PrometheusProto format.", "global.scrape_protocols", fmt.Sprintf("%v", config.DefaultGlobalConfig.ScrapeProtocols))
+			case "shift-compaction":
+				// TODO: make it different for each DB without forcing users to set external labels as we do for scrape offset as it doesn't make sense here.
```

Comment:
I don't thing it makes sense to force users to use external labels and make the offset rely on them as we do for scrape offsets.

I don't see any ID per DB we can rely on for this.

How about just generating a new one at startup?

Reply:
Probably we do want the magnitude to change depending on the minimum block size, or just to be user-configurable.

Reply:
the hostname isn't that unique especially across processes on the same machine, thus sharing its resources.

Reply:
I was thinking "hostname" inside a Docker container, which is fairly unique.
I can't imagine that many people are running multiple Prometheus on a machine outside of containers.

Reply:
But fairly unique (by default) only in the same namespace. If one creates multiple Prom pods with the same name across multiple namespaces they'll all end up having the same hostname.
I went for making the `offset range a percentage of the head's chunkrange` for now.